### PR TITLE
feat(product-selection): make profiling available for python django platform

### LINKED ISSUE #54273: Add Profiling to Onboarding & New Project creation for Python-Django

### PR DESCRIPTION
**Pull request by priscilawebdev**
This pull request introduces the "Profiling" option to the existing "Product Selection" feature. For now only available for the 'python-django' platform

closes: https://github.com/getsentry/sentry/issues/54273

### CODE REVIEW

File: static/app/components/onboarding/productSelection.tsx [hunk 1]
```diff
@@ -159,32 +217,62 @@ export function ProductSelection({
...
+          </Tooltip>
+        )}
+        {includeProfiling && (
+          <Tooltip
+            title={
+              profilingProductDisabled?.reason ?? (
+                <TooltipDescription>
+                  {t(
+                    'Enable code-level visibility in production environments, helping you swiftly detect performance bottlenecks and visualize hot paths in your code. Requires Performance Monitoring.'
+                  )}
```
Comment (priscilawebdev) [inline static/app/components/onboarding/productSelection.tsx:255]: @ale-cota do you approve this copy?

File: static/app/components/onboarding/productSelection.tsx [hunk 2]
```diff
@@ -159,32 +217,62 @@ export function ProductSelection({
...
+                  <ExternalLink href="https://docs.sentry.io/platforms/python/profiling/">
+                    {t('Read the Docs')}
+                  </ExternalLink>
```
Comment (priscilawebdev) [inline static/app/components/onboarding/productSelection.tsx:258]: we currently don't have documentation for profiling exclusively for Django. Shall we add it?

File: static/app/gettingStartedDocs/python/django.tsx [hunk 3]
```diff
@@ -2,13 +2,36 @@ import ExternalLink from 'sentry/components/links/externalLink';
...
 
 // Configuration Start
+
+const profilingConfiguration = `
+# To set a uniform sample rate
+# Set profiles_sample_rate to 1.0 to profile 100%
+# of sampled transactions.
+# We recommend adjusting this value in production,
+profiles_sample_rate=1.0,
+`;
```
Comment (priscilawebdev) [inline static/app/gettingStartedDocs/python/django.tsx:16]: In the full documentation version, there's an optional setup step for profiling. You can find it [here](https://docs.sentry.io/platforms/python/profiling/?#enable-profiling). Should we consider adding this step as well? How useful is it for new users?

```
   # Alternatively, to control sampling dynamically
    profiles_sampler=profiles_sampler
```

File: static/app/components/onboarding/gettingStartedDoc/layout.tsx [hunk 4]
```diff
@@ -38,20 +46,43 @@ export type LayoutProps = {
...
 }: LayoutProps) {
   const organization = useOrganization();
   const {isSelfHosted} = useLegacyStore(ConfigStore);
 
   const isJavaScriptPlatform =
     platformKey === 'javascript' || !!platformKey?.match('^javascript-([A-Za-z]+)$');
 
-  const displayProductSelection = !isSelfHosted && isJavaScriptPlatform;
+  const displayProductSelection =
+    !isSelfHosted && (isJavaScriptPlatform || includeProfiling);
```
Comment (matejminar) [inline static/app/components/onboarding/gettingStartedDoc/layout.tsx:85]: These one-off booleans (isJavaScriptPlatform, includeProfiling, hideSessionReplay) are no longer sustainable.
How about an object where we have SDK names as keys and an array of available products as values? If the SDK is not present (or the array is empty), that platform does not have a product selector.

Comment (priscilawebdev) [reply static/app/components/onboarding/gettingStartedDoc/layout.tsx:85]: yes, I noticed the same and wanted to change that in the same PR, but in the end I decided to work on it in a separated one. your idea is good. will implement that. thanks 🙏

Comment (matejminar) [reply static/app/components/onboarding/gettingStartedDoc/layout.tsx:85]: Let's please not ship like this and do it right in this PR 🙏

Comment (priscilawebdev) [reply static/app/components/onboarding/gettingStartedDoc/layout.tsx:85]: as we talked, will add this change in a separate PR and merge before this 😉

File: static/app/gettingStartedDocs/python/django.tsx (hunk 3 above)
Comment (priscilawebdev) [reply static/app/gettingStartedDocs/python/django.tsx:16]: cc @ale-cota

File: static/app/components/onboarding/productSelection.tsx [hunk 5]
```diff
@@ -19,11 +19,13 @@ export enum ProductSolution {
   ERROR_MONITORING = 'error-monitoring',
   PERFORMANCE_MONITORING = 'performance-monitoring',
   SESSION_REPLAY = 'session-replay',
+  PROFILING = 'profiling',
 }
 
 // This is the list of products that are available for each platform
-// Since the ProductSelection component is rendered in the onboarding flow, it is ok to have this list here
+// Since the ProductSelection component is rendered in the onboarding/project creation flow only, it is ok to have this list here
 export const platformProductAvailability = {
```
Comment (matejminar) [inline static/app/components/onboarding/productSelection.tsx:27]: nit: should we keep this alphabetically sorted? we usually do for a good overview and easy edits

**Review by matejminar — APPROVED** [review]

File: static/app/gettingStartedDocs/python/django.tsx (hunk 3 above)
Comment (ale-cota) [reply static/app/gettingStartedDocs/python/django.tsx:16]: We can leave this out. That's true also for Performance, but it's already an "advanced setting", so no need to put it in onboarding.

File: static/app/components/onboarding/productSelection.tsx [hunk 6]
```diff
@@ -249,15 +266,24 @@ export function ProductSelection({
...
             onClick={() => handleClickProduct(ProductSolution.PERFORMANCE_MONITORING)}
-            disabled={disabledProducts?.find(
-              disabledProduct =>
-                disabledProduct.product === ProductSolution.PERFORMANCE_MONITORING
-            )}
+            disabled={
+              urlProducts.includes(ProductSolution.PROFILING)
+                ? {
+                    reason: t(
+                      'Enabling Profiling requires Performance Monitoring, making it inseparable. As a result, disabling Performance Mornitoring is not possible while profiling is active.'
```
Comment (ale-cota) [inline static/app/components/onboarding/productSelection.tsx:277]:
```suggestion
                      'You must have Performance Monitoring set up to use Profiling. Disabling it is not possible while Profiling is selected.'
```

**Review by ale-cota — APPROVED** [review]

File: static/app/components/onboarding/productSelection.tsx (hunk 2 above)
Comment (ale-cota) [reply static/app/components/onboarding/productSelection.tsx]: Nah, I think linking to the python generic docs is sufficient 👍

**Pull request merged**